payjoin/src/send/error.rs

```diff
@@ -269,19 +269,23 @@ impl From<InternalCreateRequestError> for CreateRequestError {
     fn from(value: InternalCreateRequestError) -> Self { CreateRequestError(value) }
 }
 
-/// Represent an error returned by the receiver.
+/// Represent an error returned by Payjoin receiver.
 pub enum ResponseError {
-    /// `WellKnown` errors following the BIP78 spec
-    /// https://github.com/bitcoin/bips/blob/master/bip-0078.mediawiki#user-content-Receivers_well_known_errors
-    /// These errors are displayed to end users.
+    /// Errors that are defined in the [`BIP78`] spec.
     ///
-    /// The `WellKnownError` represents `errorCode` and `message`.
+    /// It is safe to display these errors to end users.
+    ///
+    /// [`BIP78`]: https://github.com/bitcoin/bips/blob/master/bip-0078.mediawiki#user-content-Receivers_well_known_errors
     WellKnown(WellKnownError),
-    /// `Unrecognized` errors are errors that are not well known and are only displayed in debug logs.
-    /// They are not displayed to end users.
+    /// Errors out of the scope of the [`BIP78`] spec.
+    ///
+    /// These errors are only displayed in debug logs.
+    ///
+    /// [`BIP78`]: https://github.com/bitcoin/bips/blob/master/bip-0078.mediawiki#user-content-Receivers_well_known_errors
     Unrecognized { error_code: String, message: String },
-    /// `Validation` errors are errors that are caused by malformed responses.
-    /// They are only displayed in debug logs.
+    /// Errors caused by malformed responses.
+    ///
+    /// These errors are only displayed in debug logs.
```

Comment:
The comment change is fine but "These errors are only displayed in debug logs" is false statement `impl Display for ResponseError` writes the `ValidationError` in the match arm for `ResponseError::Validation`

Consider fixing the Display implementation to only write a fixed string.

https://github.com/payjoin/rust-payjoin/blob/1f87e82e6b03b849ede6c8dcfcc8c2e25cabdd88/payjoin/src/send/error.rs#L354